Engine/Generic/RuleSuppression.cs

```diff
@@ -202,61 +202,56 @@ public RuleSuppression(AttributeAst attrAst, int start, int end)
                             break;
                         }
 
-                        switch (name.ArgumentName.ToLower())
+                        var argumentName = name.ArgumentName;
+                        if (argumentName.Equals("rulename", StringComparison.OrdinalIgnoreCase))
                         {
-                            case "rulename":
-                                if (!String.IsNullOrWhiteSpace(RuleName))
-                                {
-                                    Error = String.Format(Strings.NamedAndPositionalArgumentsConflictError, name);
-                                }
-
-                                RuleName = (name.Argument as StringConstantExpressionAst).Value;
-                                goto default;
-
-                            case "rulesuppressionid":
-                                if (!String.IsNullOrWhiteSpace(RuleSuppressionID))
-                                {
-                                    Error = String.Format(Strings.NamedAndPositionalArgumentsConflictError, name);
-                                }
-
-                                RuleSuppressionID = (name.Argument as StringConstantExpressionAst).Value;
-                                goto default;
-
-                            case "scope":
-                                if (!String.IsNullOrWhiteSpace(Scope))
-                                {
-                                    Error = String.Format(Strings.NamedAndPositionalArgumentsConflictError, name);
-                                }
-
-                                Scope = (name.Argument as StringConstantExpressionAst).Value;
-
-                                if (!scopeSet.Contains(Scope))
-                                {
-                                    Error = Strings.WrongScopeArgumentSuppressionAttributeError;
-                                }
+                            if (!String.IsNullOrWhiteSpace(RuleName))
+                            {
+                                Error = String.Format(Strings.NamedAndPositionalArgumentsConflictError, name);
+                            }
 
-                                goto default;
+                            RuleName = (name.Argument as StringConstantExpressionAst).Value;
+                        }
+                        else if (argumentName.Equals("rulesuppressionid", StringComparison.OrdinalIgnoreCase))
```

Comment:
There's a lot of `else if`s in this method and I had to read all of them to check that some kind of state isn't being manipulated by all of them. Would it be easier to use `return`?

Reply:
Hmm, one would have to move it into a function and other variables would need to be passed to it and there is no else, meaning the caller of it would need an `if (returnValue != null)` check. I don't think the juice is worth the squeeze here, especially for code that is not likely to be changed and this change is not adding any functionality that would justify such a refactor (any refactor also comes with the risk of a regression, I am not saying we should not change in general, but there seems to be little benefit to me here). Even if previous code is not the best, as long as it does not appear to be dangerously bad, I'd not change it too much in this instance